### PR TITLE
feat(metrics): expose sink error in grafana and factor error metrics

### LINKED ISSUE #13042: Expose sink error to grafana
See #12996 for discussion.

- [ ] #13028
- [ ] Add user sink error via metric label

### PR DESCRIPTION
**Pull request by hzxa21**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Fix #13042 

- Support passing sink error from remote sink writer back to sink executor via JNI
- Implement a collector to report error metrics to reduce the memory overhead
- Expose sink error to grafana

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by hzxa21** [comment]
image

**Review by wenym1 — APPROVED** [review]
Rest LGTM!

Can we have a picture to see how the error is displayed on grafana?

**Review by chenzl25 — APPROVED** [review]
The JNI part LGTM!

File: src/stream/src/executor/sink.rs [hunk 1]
```diff
@@ -290,8 +292,11 @@ impl<F: LogStoreFactory> SinkExecutor<F> {
         log_reader: R,
         columns: Vec<ColumnCatalog>,
         sink_writer_param: SinkWriterParam,
+        actor_context: ActorContextRef,
     ) -> StreamExecutorResult<Message> {
         let metrics = sink_writer_param.sink_metrics.clone();
+        let connector_name = sink_writer_param.connector_name.clone();
```
[the remark this answers is not in the document]
Comment (hzxa21) [reply src/stream/src/executor/sink.rs]: Good point. Done/

File: src/jni_core/src/lib.rs [hunk 2]
```diff
@@ -922,6 +923,28 @@ pub extern "system" fn Java_com_risingwave_java_binding_Binding_sendSinkWriterRe
...
 }
 
+#[no_mangle]
+pub extern "system" fn Java_com_risingwave_java_binding_Binding_sendSinkWriterErrorToChannel<'a>(
+    env: EnvParam<'a>,
+    channel: Pointer<'a, Sender<anyhow::Result<SinkWriterStreamResponse>>>,
+    msg: JString<'a>,
+) -> jboolean {
+    execute_and_catch(env, move |env| {
+        let err_msg: String = env
```
[the remark this answers is not in the document]
Comment (hzxa21) [reply src/jni_core/src/lib.rs:933]: What do you mean? I think we need to pass the error to the channel.

[the remark this answers is not in the document]
Comment (wenym1) [reply src/jni_core/src/lib.rs:933]: I mean the error from `env.get_string`.

**Pull request merged**